extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriLightHandler.java

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.tradfri.handler;
+
+import static org.eclipse.smarthome.binding.tradfri.TradfriBindingConstants.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.eclipse.smarthome.binding.tradfri.DeviceConfig;
+import org.eclipse.smarthome.binding.tradfri.internal.CoapCallback;
+import org.eclipse.smarthome.binding.tradfri.internal.TradfriCoapClient;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TradfriLightHandler} is responsible for handling commands for
+ * individual lights.
+ *
+ * @author Kai Kreuzer - Initial contribution
+ */
+public class TradfriLightHandler extends BaseThingHandler implements CoapCallback {
+
+    private final Logger logger = LoggerFactory.getLogger(TradfriLightHandler.class);
+
+    // step size for increase/decrease commands
+    private static final int STEP = 10;
+
+    // keeps track of the current state for handling of increase/decrease
+    private LightData state = null;
```

Comment:
`null` and `false` are defaults and you don't set the defaults for `id` and `coapClient`